odoo/orm/identifiers.py

```diff
@@ -38,4 +38,7 @@ def __str__(self):
         return "NewId_%s" % id_part
 
 
-IdType: typing.TypeAlias = int | NewId
+# By default, in the ORM we initialize it as an int, but any type should work.
+# However, and some parts of the ORM may assume it is an integer.
+# Non-exhaustive list: relational fields, references, hierarchies, etc.
+IdType: typing.TypeAlias = int | NewId | str
```

Comment:
Why not a tuple for instance?

Reply:
We only support fetching a single column. We could add other types like uuid and maybe extended postgres types, but I'm keeping it simple.